src/main/java/meteordevelopment/meteorclient/systems/modules/world/SpawnProofer.java

```diff
@@ -0,0 +1,187 @@
+            return;
+        }
+
+        // Find spawn locations
+        for (Spawn spawn : spawns) spawnPool.free(spawn);
+        spawns.clear();
+        BlockIterator.register(range.get(), range.get(), (blockPos, blockState) -> {
+            BlockUtils.MobSpawn spawn = BlockUtils.isValidMobSpawn(blockPos);
+            if ((spawn == BlockUtils.MobSpawn.Always && alwaysSpawns.get()) ||
+                    spawn == BlockUtils.MobSpawn.Potential && potentialSpawns.get()) spawns.add(spawnPool.get().set(blockPos));
+        });
+    }
+
+    @EventHandler
+    private void onTickPost(TickEvent.Post event) {
+        // Delay
+        if (delay.get() != 0 && ticksWaited < delay.get() - 1) {
+            ticksWaited++;
+            return;
+        }
+        if (spawns.isEmpty()) return;
+
+
+        // Find slot
+        FindItemResult block = InvUtils.findInHotbar(itemStack -> blocks.get().contains(Block.getBlockFromItem(itemStack.getItem())));
+
+        // Place blocks
+        if (delay.get() == 0) {
+            for (Spawn spawn : spawns) BlockUtils.place(spawn.blockPos, block, rotate.get(), -50, false);
+        } else {
+
+            // Check if light source
+            if (isLightSource(Block.getBlockFromItem(mc.player.getInventory().getStack(block.getSlot()).getItem()))) {
+
+                // Find lowest light level
+                int lowestLightLevel = 16;
+                Spawn selectedSpawn = spawns.get(0);
+                for (Spawn spawn : spawns) {
+                    int lightLevel = mc.world.getLightLevel(spawn.blockPos);
+                    if (lightLevel < lowestLightLevel) {
+                        lowestLightLevel = lightLevel;
+                        selectedSpawn = spawn;
+                    }
+                }
+
+                BlockUtils.place(selectedSpawn.blockPos, block, rotate.get(), -50, false);
+
+            } else {
+
+                BlockUtils.place(spawns.get(0).blockPos, block, rotate.get(), -50, false);
+
+            }
+
+        }
+
+        ticksWaited = 0;
+    }
+
+    private List<Block> getDefaultBlocks() {
+
+        ArrayList<Block> defaultBlocks = new ArrayList<>();
+        for (Block block : Registry.BLOCK) {
+            if (filterBlocks(block)) defaultBlocks.add(block);
+        }
+        return defaultBlocks;
+    }
+
+    private boolean filterBlocks(Block block) {
+        return isNonOpaqueBlock(block) || isLightSource(block);
+    }
+
+    private boolean isNonOpaqueBlock(Block block) {
+        return block instanceof AbstractButtonBlock ||
+            block instanceof SlabBlock ||
+            block instanceof AbstractPressurePlateBlock ||
+            block instanceof TransparentBlock ||
+            block instanceof TripwireBlock;
+    }
+
+    private boolean isLightSource(Block block) {
+        return block.getDefaultState().getLuminance() > 0;
+    }
+
+    private static class Spawn {
+        public BlockPos.Mutable blockPos = new BlockPos.Mutable();
```

Comment:
this class is usless, just use the BlockPos.Mutable class for your list and pool